### PR TITLE
Add LimitedWorkerPool

### PR DESCRIPTION
**Pull request by trowski**
This adds a new interface, `LimitedWorkerPool`, which extends the existing `WorkerPool` adding a method to get the maximum number of workers available in the pool.

### CODE REVIEW

File: src/Worker/DelegatingWorkerPool.php [hunk 1]
```diff
@@ -116,7 +121,7 @@ public function getWorker(): Worker
         return new PooledWorker($this->selectWorker(), $this->push(...));
     }
 
-    public function getLimit(): int
+    public function getWorkerLimit(): int
```
Comment (kelunik) [inline src/Worker/DelegatingWorkerPool.php:124]: Why did you rename this in-place? I guess it's still unreleased?

Comment (trowski) [reply src/Worker/DelegatingWorkerPool.php:124]: Correct. No release has been made with `DelegatingWorkerPool` available.

**Review by kelunik — APPROVED** [review]
